src/tests/unit/tests/reports/package/footer-text-for-service.test.tsx

```diff
@@ -1,21 +1,38 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import { EnvironmentInfo } from 'common/environment-info-provider';
+import { EnvironmentInfo, EnvironmentInfoProvider } from 'common/environment-info-provider';
 import { shallow } from 'enzyme';
 import * as React from 'react';
-import { FooterTextForService } from 'reports/package/footer-text-for-service';
+import { FooterTextForService, FooterTextForServiceProps } from 'reports/package/footer-text-for-service';
+import { IMock, Mock, Times } from 'typemoq';
 
 describe('FooterTextForService', () => {
+    let environmentInfoProviderMock: IMock<EnvironmentInfoProvider>;
+
+    const environmentInfo: EnvironmentInfo = {
+        extensionVersion: '1.2.3',
+        browserSpec: 'dummy browser version 1.0.1',
+        axeCoreVersion: '4.5.6',
+    };
+
+    beforeEach(() => {
+        environmentInfoProviderMock = Mock.ofType(EnvironmentInfoProvider);
+
+        environmentInfoProviderMock.setup(eipm => eipm.getEnvironmentInfo()).returns(() => environmentInfo).verifiable(Times.once());
+    });
+
     it('renders', () => {
-        const environmentInfo: EnvironmentInfo = {
-            extensionVersion: '1.2.3',
-            browserSpec: 'dummy browser version 1.0.1',
-            axeCoreVersion: '4.5.6',
+        const footerTextForServiceProps: FooterTextForServiceProps = {
+            environmentInfoProvider: environmentInfoProviderMock.object,
         };
 
         const FooterText = FooterTextForService('ClientService');
 
-        const footerWrapper = shallow(<FooterText {...{ environmentInfo }} />);
+        const footerWrapper = shallow(<FooterText {...footerTextForServiceProps} />);
         expect(footerWrapper.getElement()).toMatchSnapshot('footer');
     });
+
+    afterEach(() => {
```

Comment:
I know this works but if I remember correctly, we decided to not verify within the afterEach blocks?

@smoralesd can maybe add to that.

Reply:
Good catch. That's right, it's better not to break the AAA steps from the unit test, all of the steps should explicitly be part of the test itself.